x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/SearchableSnapshotDirectory.java

```diff
@@ -288,42 +294,39 @@ public final long fileLength(final String name) throws IOException {
 
     @Override
     public Set<String> getPendingDeletions() {
-        throw unsupportedException();
+        throw unsupportedException("getPendingDeletions");
     }
 
     @Override
-    public void sync(Collection<String> names) {
-        throw unsupportedException();
```

Comment:
Just so I am sure I understand this correctly, we still call `sync` and `syncMetadata` but do not modify anything? I wonder why these are still called?